src/learning/utils.py

```diff
@@ -0,0 +1,51 @@
+import torch
+import torchvision.transforms as transforms
+
+
+class CycleData():
+    def __init__(self, discr_x, discr_y, gen_xy, gen_yx, batch_x, batch_y):
+        self.batch_x = batch_x
+        self.batch_y = batch_y
+        self.synthesis_x = gen_yx(batch_y)
+        self.synthesis_y = gen_xy(batch_x)
+        self.batch_x_predictions = discr_x(self.batch_x)
+        self.batch_y_predictions = discr_y(self.batch_y)
+        self.synthesis_x_predictions = discr_x(self.synthesis_x)
+        self.synthesis_y_predictions = discr_y(self.synthesis_y)
+        self.id_x_approximations = gen_yx(self.synthesis_y)
+        self.id_y_approximations = gen_xy(self.synthesis_x)
+
+
+def get_transform(image_size):
+    return transforms.Compose([
+        transforms.Resize(image_size),
+        transforms.CenterCrop(image_size),
+        transforms.ToTensor(),
+        transforms.Normalize((0.5, 0.5, 0.5), (0.5, 0.5, 0.5))])
```

Comment:
Why are we setting the means and standard deviations to be 0.5 for all channels?

Reply:
Post hoc rationalization: We are more keen on discovering features relative to the image than relative to other images. In particular, we have no reason to treat channels differently.
Reason: Other people do it.
Does that answer your question?

Reply:
Kind of 😁